examples/external-data/src/model/containerCode.ts

```diff
@@ -69,16 +69,16 @@ export class BaseDocumentContainerRuntimeFactory extends ModelContainerRuntimeFa
 		);
 		// Register listener only once the model is fully loaded and ready
 		runtime.on("signal", (message) => {
-			if (
-				(message?.content as { type?: unknown } | undefined)?.type ===
-				SignalType.ExternalDataChanged
-			) {
-				const taskListId = (message?.content as { taskListId?: unknown } | undefined)
-					?.taskListId as string;
-				const taskList = taskListCollection.getTaskList(taskListId);
+			if (message?.type === SignalType.ExternalDataChanged) {
+				const externalTaskListId = (
+					message?.content as
+						| { externalTaskListId?: unknown; taskList?: unknown }
```

Comment:
Does the signal content have both properties? If so, we're not using the second one, but I think that one probably isn't actually there...?

Also, we're currently saying with the typing we're putting over `unknown` that `message.content` could be undefined. If that's possible we probably want to do a quick `if (externalTaskListId === undefined) { throw new Error("Signal with undefined externalTaskListId"); }` below before using `externalTaskListId`.

Reply:
> Does the signal content have both properties? If so, we're not using the second one, but I think that one probably isn't actually there...?

taskList  is there, we are passing it in, but it's true that we're not using it. I put it in there to demonstrate how it could be done but will remove it since it's confusing otherwise.